modules/pricer/src/main/java/com/opengamma/strata/pricer/capfloor/SurfaceIborCapletFloorletVolatilityBootstrapper.java

```diff
@@ -0,0 +1,206 @@
+/**
+ * Copyright (C) 2016 - present by OpenGamma Inc. and the OpenGamma group of companies
+ *
+ * Please see distribution for license.
+ */
+package com.opengamma.strata.pricer.capfloor;
+
+import static com.opengamma.strata.market.ValueType.NORMAL_VOLATILITY;
+
+import java.time.LocalDate;
+import java.time.Period;
+import java.time.ZonedDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import com.opengamma.strata.basics.ReferenceData;
+import com.opengamma.strata.basics.index.IborIndex;
+import com.opengamma.strata.collect.ArgChecker;
+import com.opengamma.strata.collect.array.DoubleArray;
+import com.opengamma.strata.collect.array.DoubleMatrix;
+import com.opengamma.strata.market.curve.Curve;
+import com.opengamma.strata.market.param.CurrencyParameterSensitivities;
+import com.opengamma.strata.market.sensitivity.PointSensitivities;
+import com.opengamma.strata.market.surface.InterpolatedNodalSurface;
+import com.opengamma.strata.market.surface.Surface;
+import com.opengamma.strata.market.surface.SurfaceMetadata;
+import com.opengamma.strata.market.surface.Surfaces;
+import com.opengamma.strata.pricer.impl.option.GenericImpliedVolatiltySolver;
+import com.opengamma.strata.pricer.option.RawOptionData;
+import com.opengamma.strata.pricer.rate.RatesProvider;
+import com.opengamma.strata.product.capfloor.ResolvedIborCapFloorLeg;
+
+/**
+ * Caplet volatilities calibration to cap volatilities based on interpolated surface.
+ * <p>
+ * The caplet volatilities are computed by bootstrapping along the expiry time dimension. 
+ * The result is an interpolated surface spanned by expiry and strike.  
+ * The position of the node points on the resultant surface corresponds to last expiry date of market caps. 
+ * The nodes should be interpolated by a local interpolation scheme along the time direction.  
+ * See {@link SurfaceIborCapletFloorletBootstrapDefinition} for detail.
```

Comment:
The name of this class has changed: 'SurfaceIborCapletFloorletVolatilityBootstrapDefinition'